src/shogun/lib/SGMatrix.h

```diff
@@ -41,6 +41,49 @@ template<class T> class SGMatrix : public SGReferencedData
 	public:
 		typedef RandomIterator<T> iterator;
 		typedef ConstRandomIterator<T> const_iterator;
+	
+		class column_iterator
+		{
+		public:
+			//Iterator traits
+			using difference_type = index_t;
+			using value_type = SGVector<T>;
+			using pointer = SGVector<T>*;
+			using reference = SGVector<T>&;
+			using iterator_category = std::random_access_iterator_tag;
+
+			column_iterator(SGMatrix<T>& mat, index_t idx = 0) : m_mat(mat), m_col_idx(idx) {}
+			column_iterator(const column_iterator& other) : m_mat(other.m_mat), m_col_idx(other.m_col_idx) {}
+			
+			column_iterator& operator++(){ ++m_col_idx;return *this; }
+			column_iterator operator++(int) { column_iterator retval(*this); ++(*this); return retval;}
+			column_iterator& operator--(){ --m_col_idx;return *this; }
+			column_iterator operator--(int) { column_iterator retval(*this); --(*this); return retval;}
+
+			bool operator==(const column_iterator& other) const { return m_mat == other.m_mat && m_col_idx == other.m_col_idx; }
+			bool operator!=(const column_iterator& other) const { return !(*this == other); }
+
+			value_type operator*(){	return m_mat.get_column(m_col_idx); }
+			
+			column_iterator &operator += (difference_type d) { m_col_idx += d; return *this; }
+			column_iterator &operator -= (difference_type d) { m_col_idx -= d; return *this; }
+
+			column_iterator operator + (difference_type d) const { return column_iterator(m_mat,m_col_idx+d); }
+			column_iterator operator - (difference_type d) const { return column_iterator(m_mat,m_col_idx-d); }
+
+			value_type operator [] (difference_type d) const { return *column_iterator(m_mat,m_col_idx+d); }
+
+			bool operator < (const column_iterator &other) const { return m_col_idx < other.m_col_idx; }
+			bool operator > (const column_iterator &other) const { return m_col_idx > other.m_col_idx; }
+			bool operator <= (const column_iterator &other) const { return m_col_idx <= other.m_col_idx; }
+			bool operator >= (const column_iterator &other) const { return m_col_idx >= other.m_col_idx; }
```

Comment:
Could you clang format this please? There is a mixture of styles here, such as spaces in operators and references on the variable vs on the type

Reply:
Sure!